src/nav/NavBarBackButton.js

```diff
@@ -20,11 +22,16 @@ import * as NavigationService from './NavigationService';
 // TODO: on iOS, show the right icon for a back button
 // TODO: on iOS, give the right label for a back button
 export default function NavBarBackButton(props: {| +color?: string |}) {
+  const _ = useContext(TranslationContext);
+
+  const accessibilityLabel = Platform.OS === 'ios' ? _('Back') : _('Navigate up');
   const { color } = props;
+
   return (
     <NavButton
       name="arrow-left"
-      accessibilityLabel="Navigate up"
+      accessibilityLabel={accessibilityLabel}
+      accessibilityRole="button"
```

Comment:
Hmm, it seems like `accessibilityRole="button"` should be decided in one of the less specific components that `NavBarBackButton` depends on:

- `Touchable`
- `NavButtonGeneral`
- `NavButton`

In particular, it sounds to me like the only right way to use `NavButtonGeneral` is where its role is a button, so it could probably be decided there, without even exposing an `accessibilityRole` prop itself. So, like this:

src/nav/NavButtonGeneral
```diff
   render() {
-    const { children, onPress, accessibilityLabel, accessibilityRole } = this.props;
+    const { children, onPress, accessibilityLabel } = this.props;
 
     return (
       <Touchable
         onPress={onPress}
         accessibilityLabel={accessibilityLabel}
-        accessibilityRole={accessibilityRole}
+        accessibilityRole="button"
       >
         <View style={this.styles.navButtonFrame}>{children}</View>
       </Touchable>
```

It's probably less certain that `Touchable` should have a "button" role. It might be a sensible default (I can't think of a way we currently use it that isn't as a button), but it's a good idea to let that be overridden.

A simple way to give `Touchable` that default would be the following:

src/common/Touchable.js
```diff
-    const { accessibilityLabel, style, onPress, onLongPress, accessibilityRole } = this.props;
+    const {
+      accessibilityLabel,
+      style,
+      onPress,
+      onLongPress,
+      accessibilityRole = 'button',
+    } = this.props;
```